File: sort/javascript/selectionSort.js
function selectionSort(arr) {
  for (let i = 0; i < arr.length - 1; i++) {
    let tmpIndex = i;
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[tmpIndex] > arr[j]) {
        tmpIndex = j;
      }
    }
    let tmp = arr[tmpIndex];
    arr[tmpIndex] = arr[i];
    arr[i] = tmp;
  }
  return arr;
}

const arr = [2, 6, 4, 3, 10, 9, 1, 5, 7, 8];
console.log(`Unsorted: ${arr}`);
console.log(`Selection Sort: ${selectionSort(arr)}`);

File: sort/python/selectionSort.py
def selectionSort(arr):
    for i in range(len(arr) - 1):
        tmpIndex = i
        for j in range(i + 1, len(arr)):
            if arr[tmpIndex] > arr[j]:
                arr[tmpIndex], arr[j] = arr[j], arr[tmpIndex]
    return arr


arr = [2, 6, 4, 3, 10, 9, 1, 5, 7, 8]
print("Unsorted:", arr)
print("Selection Sort:", selectionSort(arr))
